src/app/modules/bounce/bounce.service.ts

```diff
@@ -129,18 +137,34 @@ const logCriticalBounce = (
 export const updateBounces = async (body: ISnsNotification): Promise<void> => {
   const notification: IEmailNotification = JSON.parse(body.Message)
   // This is the crucial log statement which allows us to debug bounce-related
-  // issues, as it logs all the details about deliveries and bounces
-  logger.info(notification)
+  // issues, as it logs all the details about deliveries and bounces. Email
+  // confirmation info goes to the short-term log group so we do not store
+  // form fillers' information for too long, and everything else goes into the
+  // main log group.
+  if (
+    extractHeader(notification, EMAIL_HEADERS.emailType) ===
+    EMAIL_TYPES.emailConfirmation
+  ) {
+    shortTermLogger.info(notification)
+  } else {
+    logger.info({
+      message: 'Email notification',
+      meta: {
+        action: 'Received notification from SNS',
+        ...notification,
+      },
+    })
+  }
   const latestBounces = Bounce.fromSnsNotification(notification)
   if (!latestBounces) return
   const formId = latestBounces.formId
   const oldBounces = await Bounce.findOne({ formId })
   if (oldBounces) {
     oldBounces.merge(latestBounces, notification)
-    logCriticalBounce(oldBounces, formId, notification)
+    logCriticalBounce(oldBounces, notification)
     await oldBounces.save()
   } else {
-    logCriticalBounce(latestBounces, formId, notification)
+    logCriticalBounce(latestBounces, notification)
     await latestBounces.save()
```

Comment:
I'm not sure what is the best way to do this, but I think that updating the `hasAlarmed` key should not be the responsibility of `logCriticalBounce`

```
if (oldBounces) {
    oldBounces.merge(latestBounces, notification)
    logCriticalBounce(oldBounces, notification)
    bounceDoc.hasAlarmed = true
    await oldBounces.save()
} else {
    logCriticalBounce(latestBounces, notification)
    bounceDoc.hasAlarmed = true
    await latestBounces.save()
}
```

Reply:
Perhaps in line 162, since we should be updating hasAlarmed regardless of whether there are oldBounces

```
  if (!latestBounces) return
  const formId = latestBounces.formId
  const oldBounces = await Bounce.findOne({ formId })
>>bounceDoc.hasAlarmed = true
  if (oldBounces) {
    oldBounces.merge(latestBounces, notification)
    logCriticalBounce(oldBounces, notification)
    await oldBounces.save()
  } else {
    logCriticalBounce(latestBounces, notification)
    await latestBounces.save()
  }

```

Reply:
absolutely. this key is going to be repurposed in a subsequent PR anyway, currently it's just dangling. I've replaced this with a TODO linking to the "auto email form admin" issue.